### PR TITLE
mgr/BaseMgrModule: drop GIL for ceph_send_command

### PR DESCRIPTION
**Pull request by liewegas**
Otherwise, we can easily deadlock, since other bits of the code hold the
objecter lock and then take the GIL.

Fixes: http://tracker.ceph.com/issues/38537
Signed-off-by: Sage Weil <sage@redhat.com>

### CODE REVIEW

**Comment by liewegas** [comment]
http://pulpito.ceph.com/sage-2019-03-02_01:13:07-rados-wip-sage2-testing-2019-03-01-1553-distro-basic-smithi/

File: src/mgr/BaseMgrModule.cc [hunk 1]
```diff
@@ -139,6 +139,9 @@ ceph_send_command(BaseMgrModule *self, PyObject *args)
 
   MonCommandCompletion *command_c = new MonCommandCompletion(self->py_modules,
       completion, tag, PyThreadState_Get());
+
+  PyThreadState *tstate = PyEval_SaveThread();
```
Comment (tchaikov) [inline src/mgr/BaseMgrModule.cc]: after nautilus, we could have a `WithoutGIL`  scoped lock (or mutex) where we don't need to hold GIL. just like how we use `Gil` where GIL is required.

**Comment by tchaikov** [comment]
shall we worry about

```
2019-03-02T05:51:04.150 INFO:teuthology.orchestra.run.smithi093.stdout:[ RUN      ] ObjectStore/StoreTestSpecificAUSize.TooManyBlobsTest/2
2019-03-02T05:51:05.828 INFO:teuthology.orchestra.run.smithi093.stderr:seeding object 0
2019-03-02T05:51:05.828 INFO:teuthology.orchestra.run.smithi093.stderr:Op 0
2019-03-02T05:51:05.828 INFO:teuthology.orchestra.run.smithi093.stderr:available_objects: 1 in_flight_objects: 0 total objects: 1 in_flight 0
2019-03-02T05:51:06.124 INFO:teuthology.orchestra.run.smithi093.stderr:Op 200
2019-03-02T05:51:06.125 INFO:teuthology.orchestra.run.smithi093.stderr:available_objects: 1 in_flight_objects: 0 total objects: 1 in_flight 0
2019-03-02T05:51:06.804 INFO:teuthology.orchestra.run.smithi093.stderr:Op 400
2019-03-02T05:51:06.807 INFO:teuthology.orchestra.run.smithi093.stderr:available_objects: 1 in_flight_objects: 0 total objects: 1 in_flight 0
2019-03-02T05:51:07.665 INFO:teuthology.orchestra.run.smithi093.stderr:Op 600
2019-03-02T05:51:07.670 INFO:teuthology.orchestra.run.smithi093.stderr:available_objects: 1 in_flight_objects: 0 total objects: 1 in_flight 0
2019-03-02T05:51:08.172 INFO:teuthology.orchestra.run.smithi093.stderr:Op 800
2019-03-02T05:51:08.173 INFO:teuthology.orchestra.run.smithi093.stderr:available_objects: 1 in_flight_objects: 0 total objects: 1 in_flight 0
2019-03-02T05:51:10.677 INFO:teuthology.orchestra.run.smithi093.stdout:/build/ceph-14.1.0-283-g569f086/src/test/objectstore/store_test.cc:6267: Failure
2019-03-02T05:51:10.677 INFO:teuthology.orchestra.run.smithi093.stdout:Expected: (res_stat.allocated) <= (max_object_size), actual: 4259840 vs 4194304
2019-03-02T05:51:11.189 INFO:teuthology.orchestra.run.smithi093.stdout:==> rm -r bluestore.test_temp_dir
2019-03-02T05:51:11.256 INFO:teuthology.orchestra.run.smithi093.stdout:[  FAILED  ] ObjectStore/StoreTestSpecificAUSize.TooManyBlobsTest/2, where GetParam() =
 "bluestore" (7106 ms)
```
in http://pulpito.ceph.com/sage-2019-03-02_01:13:07-rados-wip-sage2-testing-2019-03-01-1553-distro-basic-smithi/3656125/ , albeit not related.

File: src/mgr/BaseMgrModule.cc (hunk 1 above)
Comment (liewegas) [reply src/mgr/BaseMgrModule.cc]: yes! i was thinking the same thing

**Pull request merged**